model/client.go

```diff
@@ -336,25 +342,21 @@ func (c *Client) UploadArchiveForTranslation(filename string) (string, error) {
 		return "", errors.Wrap(err, "failed to send HTTP request to AWAT")
 	}
 
-	defer func() {
-		_, _ = io.Copy(io.Discard, resp.Body)
-		_ = resp.Body.Close()
-	}()
-
-	if resp.StatusCode != http.StatusAccepted {
-		return "", errors.Errorf("received unexpected code %d from AWAT", resp.StatusCode)
-	}
-
-	archiveNameBytes, err := io.ReadAll(resp.Body)
+	bodyBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return "", errors.New("failed to read response body")
 	}
+	defer resp.Body.Close()
```

Comment:
Any reason to change this part? Previous one looks okay to me. 

Order: Do -> defer Close -> Status Check -> Read Body

Reply:
We are reading the body first so we can display the output of the http call in the error

Reply:
What about if we encounter an error while reading body? It will not be closed.

Reply:
I got what you're saying now, yeah. Was thinking about the reader from the point of reading (`io.ReadAll`) but the reader exists from the point the request is made. I've moved the deferred close calls just after the request is made (and there's no error with it).

Reply:
Should we discard the body if we fail to read it? Can you point me to that issue so I can check?

Reply:
I will try to find a reference for that.

Reply:
@fmartingr I think this will give you a hint https://stackoverflow.com/a/60931604

Reply:
So I have updated the PR with discards anyway, just in case anything fails, as #37 did. I've tried by reading the body with `io.ReadAll` and then copying it to `io.Discard` (if everything goes right) and it works without issue as well, so being safe here just in case.